dsc_lib/src/util.rs

```diff
@@ -37,3 +57,98 @@ pub fn parse_input_to_json(value: &str) -> Result<String, DscError> {
         }
     }
 }
+
+/// Will search setting files for the specified setting.
+///
+/// # Arguments
+///
+/// * `value_name` - The name of the setting.
+///
+/// # Errors
+///
+/// Will return `Err` if could not find requested setting.
+pub fn get_setting(value_name: &str) -> Result<DscSettingValue, DscError> {
+
+    const SETTINGS_FILE_NAME: &str = "settings.dsc.json";
+    // Note that default settings file name has a version that is specific to this version of dsc
+    const DEFAULT_SETTINGS_FILE_NAME: &str = "default_settings.v1.dsc.json";
+
+    let mut result: DscSettingValue = DscSettingValue::default();
+    let mut settings_file_path : PathBuf;
+
+    if let Some(exe_home) = env::current_exe()?.parent() {
+        // First, get setting from the default settings file
+        settings_file_path = exe_home.join(DEFAULT_SETTINGS_FILE_NAME);
+        if let Ok(v) = load_value_from_json(&settings_file_path, value_name) {
+            result.setting = v;
+            debug!("Found setting '{}' in {}", &value_name, settings_file_path.to_string_lossy());
+        } else {
+            debug!("Did not find setting '{}' in {}", &value_name, settings_file_path.to_string_lossy());
+        }
+
+        // Second, get setting from the active settings file overwriting previous value
+        settings_file_path = exe_home.join(SETTINGS_FILE_NAME);
+        if let Ok(v) = load_value_from_json(&settings_file_path, value_name) {
+            result.setting = v;
+            debug!("Found setting '{}' in {}", &value_name, settings_file_path.to_string_lossy());
+        } else {
+            debug!("Did not find setting '{}' in {}", &value_name, settings_file_path.to_string_lossy());
+        }
+    } else {
+        debug!("Can't get dsc executable path");
+    }
+
+    // Third, get setting from the policy
+    settings_file_path = PathBuf::from(get_settings_policy_file_path());
+    if let Ok(v) = load_value_from_json(&settings_file_path, value_name) {
+        result.policy = v;
+        debug!("Found setting '{}' in {}", &value_name, settings_file_path.to_string_lossy());
+    } else {
+        debug!("Did not find setting '{}' in {}", &value_name, settings_file_path.to_string_lossy());
+    }
+
+    if (result.setting == serde_json::Value::Null) &&
+       (result.policy == serde_json::Value::Null) {
+        return Err(DscError::NotSupported(format!("Could not find '{value_name}' in settings").to_string()));
+    }
+
+    Ok(result)
+}
+
+fn load_value_from_json(path: &PathBuf, value_name: &str) -> Result<serde_json::Value, DscError> {
```

Comment:
It looks like we read the file on every query to settings?  Ideally we should just read it once and deserialize into a struct that other things can use.  Perhaps store it in thread local storage?

Reply:
I don't think it is a problem at this point. We read it only 2 times at this point.
We have a much bigger perf problem with repeated resource manifest deserialization; I'd rather spend time on that.

Reply:
In that case, can you just add a comment in the code reflecting this?

Reply:
Added a comment to the public `get_setting` function that should be used for reading settings.